modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PageMemoryMvPartitionStorage.java

```diff
@@ -338,24 +353,38 @@ private VersionChain findVersionChainForModification(RowId rowId) {
             throw new NoUncommittedVersionException();
         }
 
-        RowVersion currentVersion = findLatestRowVersion(currentVersionChain, ALWAYS_LOAD_VALUE);
-        assert currentVersion.isUncommitted();
+        RowVersion latestVersion = findLatestRowVersion(currentVersionChain, ALWAYS_LOAD_VALUE);
```

Comment:
I believe that this logic is already changed in master, can you check? Aborting a row that had no write intent is not an exceptional situation anymore

Reply:
The current code in `main` returns `null` if transactionId is `null`, this happens before the assert you mention here, so the assert will not be triggered for 'no write intent' situation. But it probably still makes sense to leave it here (because, if there IS a write intent, it must be uncommitted, i.e. have no timestamp).

WDYT?

Reply:
Right now I can't navigate to the code from this comments branch, because you force-pushed your code and commits have changed. Why did you do this?
Anyways, I obviously missed the part where you return null, that's on me. Everything's good